components/widgets/resources/src/main/resources/resources/uicomponents/widgets/extraGeneVariantData.js

```diff
@@ -0,0 +1,524 @@
+        var geneIndex = item.className.substring(item.className.lastIndexOf('-')+1);
+        var showIcon = new Element('span', {'class' : 'fa fa-plus-square-o fa-lg'});
+        var variantShow = new Element('button', {'class' : 'tool button secondary v-collapsed', 'type' : 'button'}).insert(showIcon).insert(" $services.localization.render('PhenoTips.GeneVariantClass.expand')");
+        var hideIcon = new Element('span', {'class' : 'fa fa-minus-square-o fa-lg'});
+        var variantHide = new Element('button', {'class' : 'tool button secondary', 'type' : 'button'}).insert(hideIcon).insert(" $services.localization.render('PhenoTips.GeneVariantClass.collapse')");
+        var variantShowWrapper = new Element('span', {'class' : 'buttonwrapper'}).insert(variantShow);
+        var variantHideWrapper = new Element('span', {'class' : 'buttonwrapper'}).insert(variantHide);
+        var variantExpandTools = new Element('span', {'class' : 'expand-tools'}).insert(variantShowWrapper).insert(variantHideWrapper);
+        item.insert({bottom: variantExpandTools});
+        [variantShow, variantHide].invoke('observe', 'click', function (event) {
+          variantShow.toggleClassName('v-collapsed');
+          variantHide.toggleClassName('v-collapsed');
+          $$('.variant-hide-heading-' + geneIndex).each( function(item) {
+              if (variantShow.hasClassName('v-collapsed')) {
+                item.toggleClassName('v-collapsed', false);
+                //collapse all more info for variants
+                $$('.variant-hide-heading-' + geneIndex+'.variant-moreinfo-row').each( function(item) {
+                  item.toggleClassName('v-collapsed', true);
+                  item.previous().down('.show-moreinfo-button').toggleClassName('triRight', true);
+                  item.previous().down('.show-moreinfo-button').toggleClassName('triDown', false);
+                });
+              }
+              else
+                item.toggleClassName('v-collapsed', true);
+            });
+        });
+      });
+    },
+
+    createEditDoneButtons : function (className) {
+      $$(className).each(function(row) {
+        var variantIndex = row.className.substring(row.className.lastIndexOf('-')+1);
+        var editVariantLink = new Element('a', {'class' : 'action-edit variant-edit-done v-collapsed', 'href' : '#', id : 'PhenoTips.GeneVariantClass_'+variantIndex+'_editDone'});
+        row.insert(editVariantLink);
+      });
+    },
+
+    createEditButtons : function (className) {
+      $$(className).each(function(row) {
+        var variantIndex = row.className.substring(row.className.lastIndexOf('-')+1);
+        var editVariantLink = new Element('a', {'class' : 'action-edit button secondary variant-edit fa fa-pencil', 'href' : '#', 'title' : "$services.localization.render('PhenoTips.GeneVariantClass.edit.hint')", id : 'PhenoTips.GeneVariantClass_'+variantIndex+'_edit', 'style' : 'font-size: 90%'}).update(' Edit');
```

Comment:
Wow, there were a lot of jshint errors and warnings. Going to commit a new version shortly.